### ISSUE TITLE
TYPO3 v11 LTS compatible version?

### ISSUE THREAD

**Issue by the-andyman** [jweiland-net/service_bw2#64]
Hello and first of all many thanks for the great Service-BW extension!
Will there also be a TYPO3 v11.5 compatible version of the extension and if so, is there already an approximate schedule for it?

**Comment by froemken** [comment]
Hello @the-andyman 

I sure there will be a version for TYPO3 11, but our customer who needs that will migrate from TYPO3 10 to 12. So upgrading this extension has no priority.
Today, I have tried (just for you) to create a TYPO3 11 compatible version, but it was harder than I thought. Our extension has a relation to solr and overwrites some classes of solr. EXT:solr for TYPO3 11 was officially released just 6-7 days ago and their team has changed a lot under the hood. So, we have to adapt these changes in service_bw2, too.

I have pushed my current state here in this repository in branch typo3_11_compatibility. I currently do not know when I have further time to touch this code. You can try testing this branch, you can try to provide some PRs, and, of cause, you can contact us for an offer to speed up development.

Have a nice start into this weekend

Stefan

**Comment by the-andyman** [comment]
Thank you for creating the TYPO3 11 compatible version just for me ;)
Unfortunately, the installation via the backend crashes with numerous errors and therefore cannot be used. 
The neccessary maps2 Extension was installed before and is running.

`Core: Exception handler (WEB): Uncaught TYPO3 Exception: Expected to find class "JWeiland\Maps2\Client\AbstractClient" in file "/html/typo3/typo3conf/ext/maps2/Classes/Client/AbstractClient.php" while importing services from resource "../Classes/*", but it was not found! Check the namespace prefix used with the resource. | Symfony\Component\DependencyInjection\Exception\InvalidArgumentException thrown in file /html/typo3/typo3_src-11.5.15/vendor/symfony/dependency-injection/Loader/FileLoader.php in line 224. Requested URL: http://servicebw.demowebsite.de/typo3/ajax/system-information/render?token=--AnonymizedToken--&skipSessionUpdate=1`

`Core: Exception handler (WEB): Uncaught TYPO3 Exception: Expected to find class "JWeiland\Maps2\Client\AbstractClient" in file "/html/typo3/typo3conf/ext/maps2/Classes/Client/AbstractClient.php" while importing services from resource "../Classes/*", but it was not found! Check the namespace prefix used with the resource. | Symfony\Component\DependencyInjection\Exception\InvalidArgumentException thrown in file /html/typo3/typo3_src-11.5.15/vendor/symfony/dependency-injection/Loader/FileLoader.php in line 224. Requested URL: http://servicebw.demowebsite.de/typo3/module/tools/ExtensionmanagerExtensionmanager?token=--AnonymizedToken--`

`Core: Exception handler (WEB): Uncaught TYPO3 Exception: Expected to find class "JWeiland\Maps2\Client\AbstractClient" in file "/html/typo3/typo3conf/ext/maps2/Classes/Client/AbstractClient.php" while importing services from resource "../Classes/*", but it was not found! Check the namespace prefix used with the resource. | Symfony\Component\DependencyInjection\Exception\InvalidArgumentException thrown in file /html/typo3/typo3_src-11.5.15/vendor/symfony/dependency-injection/Loader/FileLoader.php in line 224. Requested URL: http://servicebw.demowebsite.de/typo3/module/tools/toolsmaintenance?token=--AnonymizedToken--`

`Core: Exception handler (WEB): Uncaught TYPO3 Exception: Expected to find class "JWeiland\Maps2\Client\AbstractClient" in file "/html/typo3/typo3conf/ext/maps2/Classes/Client/AbstractClient.php" while importing services from resource "../Classes/*", but it was not found! Check the namespace prefix used with the resource. | Symfony\Component\DependencyInjection\Exception\InvalidArgumentException thrown in file /html/typo3/typo3_src-11.5.15/vendor/symfony/dependency-injection/Loader/FileLoader.php in line 224. Requested URL: http://servicebw.demowebsite.de/typo3/record/commit?token=--AnonymizedToken--&cacheCmd=all`

`Core: Exception handler (WEB): Uncaught TYPO3 Exception: Expected to find class "JWeiland\Maps2\Client\AbstractClient" in file "/html/typo3/typo3conf/ext/maps2/Classes/Client/AbstractClient.php" while importing services from resource "../Classes/*", but it was not found! Check the namespace prefix used with the resource. | Symfony\Component\DependencyInjection\Exception\InvalidArgumentException thrown in file /html/typo3/typo3_src-11.5.15/vendor/symfony/dependency-injection/Loader/FileLoader.php in line 224. Requested URL: http://servicebw.demowebsite.de/typo3/module/tools/ExtensionmanagerExtensionmanager?token=--AnonymizedToken--`

`Core: Exception handler (WEB): Uncaught TYPO3 Exception: Class "ApacheSolrForTypo3\Solr\IndexQueue\Indexer" not found while loading "JWeiland\ServiceBw2\Indexer\Indexer". | Symfony\Component\DependencyInjection\Exception\RuntimeException thrown in file /html/typo3/typo3_src-11.5.15/vendor/symfony/dependency-injection/Compiler/DefinitionErrorExceptionPass.php in line 54. Requested URL: http://servicebw.demowebsite.de/typo3/module/tools/ExtensionmanagerExtensionmanager?token=--AnonymizedToken--&tx_extensionmanager_tools_extensionmanagerextensionmanager%5BextensionKey%5D=service_bw2&tx_extensionmanager_tools_extensionmanagerextensionmanager%5Baction%5D=toggleExtensionInstallationState&tx_extensionmanager_tools_extensionmanagerextensionmanager%5Bcontroller%5D=Action`

**Comment by froemken** [comment]
Hello @the-andyman 

the most error are about `JWeiland\Maps2\Client\AbstractClient`. Our maps2 extension is a very stable extension and contains a lot of UnitTests. Seems that you have not cleared the Autoloader Cache of your TYPO3 system. Please visit Installtool and click in section "Maintenance" the button at "Rebuild PHP Autoload Information". And maybe an additional click on "Flush Cache".
With "Flush Cache" the Cache of Dependency Injection will be re-created. Only this button in Installtool will re-create the DI cache. The two Clear Cache buttons in backend at upper right won't touch this specific cache!

Stefan

**Comment by the-andyman** [comment]
Hello @froemken, 

Thank you very much for your efforts.
Unfortunately, the installation still breaks and the backend is no longer accessible. 
The current github version from today was tested on a new empty TYPO3 11 installation.

Andy

**Comment by froemken** [comment]
Good morning @the-andyman 

I can reproduce this problem on a fresh standalone TYPO3 11 installation. I have added a special configuration for Dependency Injection if EXT:solr is loaded. Else, the config will not be loaded and hopefully the error disapears.

On another TYPO3 11 installation I had to deactivate all local installed TYPO3 extensions in PackageStates.php first and activate them in ExtensionManager again. But that was tested with previous version of service_bw2. Hopefully you don't need to do this step.

Have a cool start into this short week.

Stefan

**Comment by the-andyman** [comment]
Hello @froemken,

thanks for your renewed effort!

The installation has now worked and the "Services" and "Life situations" also work so far. 
Only with the "Organizational units" comes when calling the detailed view the following error:

```
Core: Exception handler (WEB): Uncaught TYPO3 Exception: 
Typed property JWeiland\ServiceBw2\ViewHelpers\OrganisationseinheitPoiCollectionUidViewHelper::$maps2Pid must be int, string used | 
TypeError thrown in file /html/typo3/typo3conf/ext/service_bw2/Classes/ViewHelpers/OrganisationseinheitPoiCollectionUidViewHelper.php in line 59. 
```

The maps2 Extension is installed and its static template is included. 

If i want to go without the Solr extension and use another search for it, does that also work with the indexed_search, for example?

I would have, regardless of this, also a feature request. 
Can I post this here officially, or is it not intended to add more features to the extension?

**Comment by sfroemkenjw** [comment]
Hello @the-andyman 

I have just released a TYPO3 compatible version 6.0.0 to TER and Packagist.

I have solved the error in the ViewHelper. BUT: It's more a configuration problem: The Storage Page UID could not be resolved. You can read more about it in Documentation:

https://docs.typo3.org/p/jweiland/service-bw2/6.0/en-us/Configuration/TypoScript/Index.html#maps2

If it is not set service_bw2 will not show any map position

Stefan

**Comment by sfroemkenjw** [comment]
@the-andyman 

Of cause you can create feature request.

If you need an Event (new version of Hooks) in service_bw2 I'm sure I can implement that very shortly.
If you have a bigger feature, one of my colleagues from product management can create an offer for you.

Stefan
